File: src/PhotoGallery/Entities/Product.cs
namespace PhotoGallery.Entities
{
    using System;
    using System.Collections.Generic;
    
    public class Product : IEntityBase
    {
        public Product()
        {
            this.OrderLines = new HashSet<OrderLine>();
        }
    
        public int Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public Nullable<decimal> Price { get; set; }
        public string Barcode { get; set; }
        public Nullable<int> StoreId { get; set; }
        public string PicUrl { get; set; }
        public virtual ICollection<OrderLine> OrderLines { get; set; }
        public virtual Store Store { get; set; }
    }
}

File: src/PhotoGallery/wwwroot/app/core/domain/product.ts
export class Product {
    SkuId: number;
    SkuCode: string;
    SkuName: string;
    SkuPrice: number;
    BarCode: string;
    StoreId: string;
    StoreName: string;
    PicUrl: string;

    constructor(Id: number,
        Code: string,
        Name: string,
        Price: number,
        Barcode: string,
        StoreId: string,
        StoreName: string,
        PicUrl: string) {
        this.SkuId = Id;
        this.SkuCode = Code;
        this.SkuName = Name;
        this.SkuPrice = Price;
        this.BarCode = Barcode;
        this.StoreId = StoreId;
        this.StoreName = StoreName;
        this.PicUrl = PicUrl;
    }
}